src/services/expensesService.js

```diff
@@ -0,0 +1,56 @@
+let expenses = [];
+let nextId = 1;
+
+const resetExpenses = () => {
+  expenses = [];
+  nextId = 1;
+};
+
+const getAllExpenses = (userId, categories, from, to) => {
+  return expenses
+    .filter((expense) => !userId || expense.userId === +userId)
+    .filter((expense) => !categories || expense.category === categories)
+    .filter((expense) => !from || new Date(expense.spentAt) >= new Date(from))
+    .filter((expense) => !to || new Date(expense.spentAt) <= new Date(to));
+};
+
+const getExpenseById = (id) => {
+  return expenses.find((expense) => expense.id === +id);
+};
+
+const createExpense = (userId, spentAt, title, amount, category, note) => {
+  const expense = {
+    id: nextId++,
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  expenses.push(expense);
+
+  return expense;
+};
+
+const updateExpense = (id, data) => {
```

Comment:
The function 'updateExpense' directly modifies the 'expense' object without any validation of the 'data' parameter. This can lead to potential security issues and bugs. Consider validating 'data' before using it.